apps/vscode/src/providers/editor/editor.ts

```diff
@@ -140,18 +150,38 @@ export class VisualEditorProvider implements CustomTextEditorProvider {
     }));
 
     // when the active editor changes see if we have a visual editor position for it
-    context.subscriptions.push(window.onDidChangeActiveTextEditor(debounce(() => {
-      // resolve active editor
+    context.subscriptions.push(window.onDidChangeActiveTextEditor(debounce(async () => {
+
+      // resolve if active editor is text or visual
       const editor = window.activeTextEditor;
-      if (!editor) {
-        return;
-      }
-      const document = editor.document;
-      if (document && isQuartoDoc(document)) {
-        const uri = document.uri.toString();
+      const visualEditor = VisualEditorProvider.activeEditor();
+
+      let document = editor?.document || visualEditor?.document;
```

Comment:
unfortunately, visual editors are not considered editors, so when we get the `onDidChangeActiveTextEditor` notification when opening a visual editor, the `editor` value is always `undefined`. I pulled the lines you're highlighting in the latest commit, since it is not providing anything useful.

This gets at the one bit that is a bit awkward: if the default is to open in a visual editor, this won't ever fire since vscode doesn't think there is ever an active editor. So if people have the default as source but put visual in a `*.qmd` file or `_quarto.yml` file, all will act as expected. BUT if you update the editor associations to visual, files will always open in the visual editor (you can still swap to source manually) since it doesn't fire `onDidChangeActiveEditor` changing between files opened in visual editors. 

An option to avoid this behavior is to use `onDidOpenTextDocument`, but then closing editors is unpredictable since there is only the ability to close active editors, not by name. I sort of think the current setup this is okay, since my guess is most people are setting editor mode in the qmd or `_quarto.yml` file?